package/libs/openssl/Makefile

```diff
@@ -251,20 +390,39 @@ define Build/InstallDev
 endef
 
 define Package/libopenssl/install
+	$(INSTALL_DIR) $(1)/etc/ssl
+	$(CP) $(PKG_INSTALL_DIR)/etc/ssl/openssl.cnf $(1)/etc/ssl/
+	$(INSTALL_DIR) $(1)/etc/ssl/certs
+	$(INSTALL_DIR) $(1)/etc/ssl/private
+	chmod 0700 $(1)/etc/ssl/private
```

Comment:
There is an API for loading configuration from openssl.cnf from any application.  See `OPENSSL_config(3)` for openssl-1.0.2, and `OPENSSL_init_crypto(3)` for 1.1.x.  It is also used to configure engines (most of the uses outside of the utility).  Sure, the library will work fine for most people without the file, and in all but few of the cases it is used only for certificate generation with the utility, but "pedantically", it should belong to the library.

Reply:
For Ubuntu, they package openssl.cnf with the utility.

https://packages.ubuntu.com/cosmic/amd64/openssl/filelist
https://packages.ubuntu.com/cosmic/amd64/libssl1.1/filelist

Reply:
Like I pointed out earlier, technically, it belongs to the library.  Since you can't install the util without the library, it works for all cases.  It may be **required** only in corner cases--engine configuration is one of them.  
To abandon the idea of installing where it belongs, in favor of being more practical, the questions I raise are:
1. Do we want to leave out the corner cases?  
I imagine users needing to edit openssl.cnf should have the knowledge to do so, as the default version will do them no good anyway, and documentation is very sparse.
Installing it with the library makes life easier for these folks when it comes to upgrades, without having to remember to add a file to `/etc/sysupgrade.conf` (I for one am terrible at forgetting to do that).
2. What's wrong installing the conf file with the library?
A package increase of 3.5K.  Does it matter?
So I do see the point of installing it with the utility, but I still believe the right thing to do is to install it with the library, despite of what other distros are doing.

Reply:
Moving the config file to the library package will prevent multiple libopenssl binary packages to coexist. With the upcoming buildroot abi tracking changes in preparation for this and other core library bumps, binary library package will contain their ABI_VERSION as part of the package name, e.g. `libopenssl_*.ipk` becomes `libopenssl1.0.0_*.ipk` and after this update gets merged, will become `libopenssl1.1_*.ipk`. Both packages can coexist due to the bundled `*.so` having different soversions (`libssl.so.1.0.0` vs. `libssl.so.1.1`).

Shipping a common `/etc/ssl/openssl.cnf` will break this transitional mechanism as opkg will refuse to install `libopenssl1.1` over `libopenssl1.0.0` due to data file clashes.

If you think we should bundle the `openssl.cnf` nonetheless, then it should include the `ABI_VERSION` as part of the filename and we probably need to patch the library source to lookup the right one.

`libssl.so.1.0.0` needs to use something like `openssl-1.0.0.cnf` then and `libssl.so.1.1` needs to use `openssl-1.1.cnf`. Its probably simpler to just ship it with `openssl-util`, thats likely why other distros do it this way.

Reply:
I rule out patching openssl, since the `OPENSSL_CONFIG` var can be used for that purpose, but to use a non-standard path is not really ideal. How about adding a package for `openssl.cnf`?    It might not be as clean as adding to `openssl-util`, but it seems like the right thing to do IMO.  Then packages actually needing the file would add it as dependency: `openssl-util` and `afalg`, currently.  I'm working on building the `/dev/crypto` engine dynamic so we can have it as a separate package as well. I'll wait for openssl/openssl#7859 to be merged, and then backport it, as it will only be included in openssl 3.0.